commerce/domain/promotion/cart/calculator/classes/PromotionCalculatorSample.cls

```diff
@@ -62,39 +67,78 @@ public with sharing class PromotionCalculatorSample extends CartExtension.Promot
         }
     }
 
+    /**
+     * @description Evaluate BOGO promotion (Buy 5 items of qualifying product, get $2 off a unit of target product,
```

Comment:
I'm trying to understand why this use-case is not optimizable with `BuyerActionDetails`. Basically, you know the Qualifier Product Id and Target Product Id. So if `BuyerActionDetails.cartItemChanges.changedItem.ProductId` is not any of those - you can just skip the request and save time of iterating over the whole cart and resetting and applying the same promotions. Am I missing anything?

Reply:
In real world, there can be more than 1 promotion that needs to be evaluated based on store or buyer group. Each applicable promotion can have different qualifiers/targets associated. So, it is required to access all cart items any time promotions are evaluated. @curtwalker1

Reply:
@curtwalker1 correct, if buyer added qualifying product or target product - you recalculate the whole cart. But if buyer added **any other product** - then you should really recalculate nothing.

Reply:
@pplat23 Just like mentioned before, this is one of many examples. There could be multiple promotions that need to be evaluated. One promotion may require specified qualifier/target product and other may require newly added product. It could be that with change of cartItem a previously applied promotion no longer applies and a different one does. We would only know which promotions to apply when we go through all items in cart.

Reply:
To note here, this work is moving to next sprint and will be wrapped up post 252 FF

Reply:
@pplat23, @curtwalker1 Please review changes including BuyerActionDetails in [validateBuyerActionDetailsAndEvaluatePromotion()](https://github.com/forcedotcom/commerce-extensibility/pull/55/files#diff-57ce985dc4beef4e1e0eeddc5552cf5e7130ab950340372d55b0965cbd9c1400R27) method above.